src/api/app/mixins/parse_package_diff.rb

```diff
@@ -30,9 +30,12 @@ def parse_one_diff(sourcediff)
                  else # in case of deleted files
                    file['old']['name']
                  end
+      files_hash[filename] = file
       if filename.include?('/')
         other_file_keys << filename
-      elsif filename.ends_with?('.spec')
+        next
```

Comment:
Not really.

I moved the `files_hash[filename] = file` line before the conditionals. This line is always executed.

Then, I splitted the previous if in two: one for the `.include?` conditional and one for the `.ends_with?` conditionals. That is what makes the `Lint/DuplicateBranch` offense to go away.

The `next` is needed after splitting the `if`s. Otherwise, it could be that `other_file_keys << filename` will be executed twice: once inside the `if filename.include?('/')` condition, and again inside the `else` clause. And we don't want this `other_file_keys << filename` code to be run twice.

Reply:
alright, I got confused here